specs/PermissionManagement.md

```diff
@@ -0,0 +1,554 @@
+                                }
+                                return S_OK;
+                            })
+                            .Get()));
+                    return S_OK;
+                })
+                .Get(),
+            &m_DOMContentLoadedToken));
+    }
+
+    // Called when the user wants to change permission state from the custom
+    // permission management page.
+    CHECK_FAILURE(m_webView->add_WebMessageReceived(
+        Microsoft::WRL::Callback<ICoreWebView2WebMessageReceivedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2WebMessageReceivedEventArgs* args)
+                -> HRESULT
+            {
+                wil::unique_cotaskmem_string source;
+                CHECK_FAILURE(args->get_Source(&source));
+                // Permission management APIs are only used on the app's
+                // permission management page.
+                if (source.get() != m_sampleUri)
+                {
+                    return S_OK;
+                }
+                wil::unique_cotaskmem_string message;
+                CHECK_FAILURE(args->TryGetWebMessageAsString(&message));
+                // The app's permission management page can provide a way for the
+                // end user to change permissions. For example, the page can have a
+                // `Set Permission` button that triggers a dialog, where the user
+                // specifies the desired origin, permission kind, and state.
+                if (wcscmp(message.get(), L"SetPermission") == 0)
+                {
+                    m_appWindow->RunAsync([this] { ShowSetPermissionDialog(); });
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_webMessageReceivedToken));
+}
+
+void ScenarioPermissionManagement::ShowSetPermissionDialog()
+{
+    PermissionDialog dialog(m_appWindow->GetMainWindow());
+    if (dialog.confirmed && m_webViewProfile6)
+    {
+        // Example: m_webViewProfile6->SetPermissionState(
+        //    COREWEBVIEW2_PERMISSION_KIND_GEOLOCATION,
+        //    L"https://example.com",
+        //    COREWEBVIEW2_PERMISSION_STATE_DENY,
+        //    SetPermissionStateCallback);
+        CHECK_FAILURE(m_webViewProfile6->SetPermissionState(
+            dialog.kind, dialog.origin.c_str(), dialog.state,
+            Callback<ICoreWebView2StagingSetPermissionStateCompletedHandler>(
+            [this](HRESULT error) -> HRESULT
+            {
+                // Reload the permission management page.
+                m_webView->Reload();
+                return S_OK;
+            })
+            .Get()));
+    }
+}
+```
+# API Details
+
+```
+[v1_enum]
+typedef enum COREWEBVIEW2_PERMISSION_KIND {
+
+  // Other permission kinds not shown.
+
+  /// Indicates permission to automatically download multiple files. Permission
+  /// is requested when multiple downloads are triggered at the same time.
+  COREWEBVIEW2_PERMISSION_KIND_AUTOMATIC_DOWNLOADS,
```

Comment:
Is this specific to _multiple_ downloads? Name doesn't set that, but comment suggests it